### ISSUE TITLE
How to control the style or  intonation  of the synthesized speech?

### ISSUE THREAD

**Issue by NewEricWang** [NVIDIA/tacotron2#58]
Hi, I found the synthesized speech files are different by running inference many times. Their style or intonation always are diverse. How to contral the style or  intonation  of the synthesized speech?

Another question:
      How to choose the best model from a series of checkpoint models? I found it is not sure that more iteration step is corresponding to better synthesis performance.

**Comment by rafaelvalle** [comment]
The variation comes from dropout on the Prenet layer. One can train without dropout or set training to False here https://github.com/NVIDIA/tacotron2/blob/master/model.py#L99.

Assuming your validation loss plateaus, any checkpoint on that plateau area.

**Comment by NewEricWang** [comment]
@rafaelvalle , Thank you!
I get constant speech for many times by setting dropout=0.0. But the speech quality is awful。 Maybe I should train model by setting dropout=0.0 again. 

In my case, I found the learned attention can't be hold along iteration increasing. For example, I can get good speech at checkpoint 80k steps, but I can't get better speech at checkpoint 85k step, even more awful speech. 

I try to build Chinese synthesis model by tacotron2. The tone of the speech I got is very strange.
This is a raw speech:
[post_r_8001_1.wav.zip](https://github.com/NVIDIA/tacotron2/files/2191053/post_r_8001_1.wav.zip)

This is a corresponding synthesized speech: 
[test.chn.61k.r1.wav.zip](https://github.com/NVIDIA/tacotron2/files/2191052/test.chn.61k.r1.wav.zip)

**Comment by rafaelvalle** [comment]
Attention is very important for proper synthesis: a low validation score with bad alignment won't produce good samples during inference time when we're not teacher forcing, i.e. providing the real previous mel frame to the decoder.

Audio quality is extremely dependent on the mel2audio decoder. 
If by tone you mean the robotic sound, you can improve it by using a Wavenet Vocoder.
If you mean something else, please explain what it is and we can work together on it. I don't speak Chinese and, except the robotic tone from using Griffin-Lim, I think your sample sounds pretty good!

**Comment by NewEricWang** [comment]
@rafaelvalle ,Thank you very much. When synthesizing with setting dropout=0, the tones of synthesized sentences using different training-step models  are different. 
This is a sound synthesized by model with 126k-step:
[test.126k.v1.wav.zip](https://github.com/NVIDIA/tacotron2/files/2217718/test.126k.v1.wav.zip)

This is a sound synthesized by model with 122k-step:
[test.122k.v1.1.wav.zip](https://github.com/NVIDIA/tacotron2/files/2217721/test.122k.v1.1.wav.zip)

**Comment by rafaelvalle** [comment]
Closing due to inactivity.

**Comment by aidosRepoint** [comment]
> Hi, I found the synthesized speech files are different by running inference many times. Their style or intonation always are diverse. How to contral the style or intonation of the synthesized speech?
> 
> Another question: How to choose the best model from a series of checkpoint models? I found it is not sure that more iteration step is corresponding to better synthesis performance.

Hi!
For those who are interested in the same issue of getting various results on several inference iterations:
it seems that instead of using
`_ = model.cuda().eval().half()`
you can use
```
_ = model.cuda().eval().half()
__ = model.decoder.prenet.eval()
```
and this will provide you with the same results everytime
